server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/http/AuthenticationReactiveFilter.java

```diff
@@ -16,13 +16,39 @@
  * specific language governing permissions and limitations      *
  * under the License.                                           *
  ****************************************************************/
-package org.apache.james.jmap.draft;
+package org.apache.james.jmap.http;
 
-import javax.servlet.http.HttpServletRequest;
+import java.util.List;
 
+import javax.inject.Inject;
+
+import org.apache.james.jmap.draft.exceptions.UnauthorizedException;
 import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.metrics.api.MetricFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.server.HttpServerRequest;
+
+public class AuthenticationReactiveFilter {
+    private final List<AuthenticationStrategy> authMethods;
+    private final MetricFactory metricFactory;
 
-public interface AuthenticationStrategy {
+    @Inject
+    @VisibleForTesting
+    AuthenticationReactiveFilter(List<AuthenticationStrategy> authMethods, MetricFactory metricFactory) {
+        this.authMethods = authMethods;
+        this.metricFactory = metricFactory;
+    }
 
-    MailboxSession createMailboxSession(HttpServletRequest httpRequest);
+    public Mono<MailboxSession> authenticate(HttpServerRequest request) {
+        return Mono.from(metricFactory.runPublishingTimerMetric("JMAP-authentication-filter",
+            Flux.fromStream(authMethods.stream())
+                .flatMap(auth -> auth.createMailboxSession(request))
+                .onErrorContinue((throwable, nothing) -> { })
```

Comment:
upon auth error, authMethods should return empty.

This means that the failure is unexpected. We should then ERROR log the exeption.